File: ios/Plugin/ZendeskPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(ZendeskPlugin, "Zendesk",
           CAP_PLUGIN_METHOD(initialize, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(initChat, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setIdentity, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setVisitorInfo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setChatConfiguration, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setPrimaryColor, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setNotificationToken, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(showHelpCenter, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(startChat, CAPPluginReturnPromise);
)

File: ios/Plugin/ZendeskPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ZendeskPlugin)
public class ZendeskPlugin: CAPPlugin {
    private let implementation = ZendeskImpl()

    @objc func initialize(_ call: CAPPluginCall) {
        let channelKey = call.getString("key") ?? ""

        let success = implementation.initialize(channelKey)

        call.resolve()
    }

    @objc func startChat(_ call: CAPPluginCall) {
        if let vc = self.bridge?.viewController {
            implementation.startChat(parentVC: vc)
            call.resolve()
        } else {
            call.reject("No view controller available")
        }
    }
}
